index.html

```diff
@@ -27,6 +27,45 @@
 
         <!-- <script src="lib/mespeak.js"></script> -->
 
+        <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+
+        <script type="text/javascript">
+        function doSearch() {
+            const query = document.getElementById("search-box").value.toLowerCase();
```

Comment:
This line is showing an error in my console (when I run locally... haven't tried from the server yet).

Reply:
But search works now and the doSearch not found error is gone!!! nice.

Reply:
yes sir i have also faced that error in my console but i try to solve that issue  .then doSearch does not work but error goes :(

Reply:
Did you try checking to see if value is null before running toLowerCase()?

Reply:
yes sir i tried it    if (!resultsList) {
          console.error("Could not find results list element.");
          return;
        }
but doSearch stop working and error gets resolved

Reply:
I think the problem is before resultsList. Try something like this:
            const value = document.getElementById("search-box").value;
            let query = "";
            if (value !== null) {
		query = value.toLowerCase();
            }

Reply:
Actually, the search-box element is not found yet, so you cannot yet access the value attribute.
So maybe:
    const el = document.getElementById("search-box")
    let query = "";
    if (el !== null) {
        query = el.value.toLowerCase();
    }

Reply:
Sir I have tried both code but sitll my console is showing same error (index):41 Uncaught TypeError: Cannot set properties of null (setting 'innerHTML')
    at doSearch ((index):41:35)
    at HTMLDocument.<anonymous> ((index):153:17)
    at fire (jquery-2.1.4.js:194:299)
    at Object.fireWith [as resolveWith] (jquery-2.1.4.js:199:198)
    at Function.ready (jquery-2.1.4.js:206:11)
    at HTMLDocument.completed (jquery-2.1.4.js:206:291)

Thanks and Regards

Reply:
That should fix the error I saw, but then you run into the resultsList error you mentioned.
Maybe try making the element if it doesn't already exist?

Reply:
Yes sir when i tried to modify my code of doSearch function for the null case the error gets resolved in the console but the reults does not show.


Thanks and Regards

Reply:
I am not sure what to do... if you create the missing element, search-box, you'll need to also populate it with the required attributes, like value. It'll take a bit of digging to ensure it is created properly.

Reply:
Ohh :(

Reply:
Sir i have solved the null issue by using if conditions and my search is still working  but i am facing a new issue in my console regarding blockname .blockname are defined in which file sir 
(index):49 Uncaught ReferenceError: blockIndex is not defined
    at doSearch ((index):49:37)
    at HTMLDocument.<anonymous> ((index):154:17)
    at fire (jquery-2.1.4.js:194:299)
    at Object.fireWith [as resolveWith] (jquery-2.1.4.js:199:198)
    at Function.ready (jquery-2.1.4.js:206:11)
    at HTMLDocument.completed (jquery-2.1.4.js:206:291)

Thanks and Regards